### ISSUE TITLE
NameError: name 'cmd2' is not defined for `cmd1; cmd2 && cmd3`, but `cmd1; cmd2 arg && cmd3` works

### ISSUE THREAD

**Issue by MrMetric** [xonsh/xonsh#2584]
These commands:
```
$ clear; make && ./program
$ clear && make && ./program
```
result in this error: `NameError: name 'make' is not defined`. These commands work, but none of them do what I want:
```
$ clear; make
$ clear; make; ./program
$ clear && make
$ clear && make; ./program
```
Note: a workaround is `clear; ![make] && ./program`.

I found that this has nothing to do with clearing the screen, running `make`, or running my program. These do not work:
```
$ echo; echo && echo
$ echo; echo && echo a
$ echo a; echo && echo
$ echo a; echo && echo a
```
but these do:
```
$ echo; echo a && echo
$ echo; echo a && echo a
$ echo a; echo a && echo
$ echo a; echo a && echo a
```

`xonfig` output:
```
+------------------+----------------------+
| xonsh            | 0.6.0.dev41          |
| Git SHA          | fa61b69a             |
| Commit Date      | Jan 13 13:37:39 2018 |
| Python           | 3.6.4                |
| PLY              | 3.9                  |
| have readline    | True                 |
| prompt toolkit   | 1.0.15               |
| shell type       | prompt_toolkit       |
| pygments         | 2.2.0                |
| on posix         | True                 |
| on linux         | True                 |
| distro           | arch                 |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+```

**Comment by astronouth7303** [comment]
Sounds a lot like #2430

**Comment by gforsyth** [comment]
Closed in #2845
